ecs-cli/modules/cli/configure/configure.go

```diff
@@ -14,40 +14,63 @@
 package configure
 
 import (
+	"errors"
+
 	"github.com/Sirupsen/logrus"
 	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands"
 	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
 	"github.com/urfave/cli"
 )
 
-// ConfigureCluster is the callback for ConfigureCommand (cluster).
-func ConfigureCluster(context *cli.Context) {
-	// get relevant fields
+func fieldEmpty(flag string, context *cli.Context) bool {
+	field := context.String(flag)
+	if field == "" {
+		logrus.Errorf("%s can not be empty.", flag)
+		return true
+	}
+	return false
+}
+
+// Cluster is the callback for ConfigureCommand (cluster).
+// Note: The error returned is only used in the unit tests
+// This function is called by urfave/cli which does not check the error returned
+func Cluster(context *cli.Context) error {
+	// validate fields not empty
+	if fieldEmpty(command.RegionFlag, context) || fieldEmpty(command.ConfigNameFlag, context) || fieldEmpty(command.ClusterFlag, context) {
+		// fieldEmpty() will log the error; returned error is only used for unit tests
```

Comment:
The configure commands have to print the error for the user; these functions are called by urfave/cli as the "Action" in the command- and the return argument will be ignored. The returned error is only used in the unit tests to test error conditions. The message in the returned error does not actually matter. This code may be a bit confusing, but its the cleanest way to make the code easily unit testable.